### ISSUE TITLE
Resuming live stream fails. (dev branch)

### ISSUE THREAD

**Issue by ogzpub** [videojs/videojs-contrib-hls#446]
If I pause a live stream and wait a little while before resuming I'm unable to continue viewing. It plays for a few seconds, presumably the remainder of the segment it was on, and then stops. In the network activity however I can see it downloading every single segment between when I paused and resumed. Should it not simply jump to a few segments before the most recent and download/play from there?

**Comment by Tralapo** [comment]
I used to use the code below with the default video.JS (so without this contrib-hls) to make sure it would always load the most recent part, both for RTMP on desktop and HLS on mobile.

```javascript
function yourSweetReadyFunc() {
  var player = this;
  player.on("pause", function () {
    player.one("play", function () {
      player.load();
      player.play();
    });
  });
};

var myPlayer = videojs('my_video_1', {}, yourSweetReadyFunc);
```
But with contrib-HLS, this code no longer works. Maybe someone could provide help with that? It would fix the problem you describe.

**Comment by ogzpub** [comment]
I've been using something similar to that for RTMP as well with good success. I believe they said once they weren't going to fix RTMP since it wasn't the focus of the project and the fix works well enough. So I'm ok with continuing to use that for RTMP.

With HLS I imagine I'll probably end up coding a similar workaround for now but I think it would be nice to have it work out of the box instead if possible since it is the project's focus time time.

**Comment by Tralapo** [comment]
Even YouTube does that with livestreams, so I guess it's kind of a standard to start again where you stopped before. I don't think it's a logical thing, but a lot of people seem to think it is.

I'm perfectly fine with using a workaround. Do you have any idea how to change mine above to make it work for contrib-hls? It works on places (mobile) where hls is native, but it breaks with contrib-hls in desktop browsers.

**Comment by dmlap** [comment]
@ogzpub: We're aware of an issue in 1.2.1 where if you pause too long, the player will not be able to resume a live stream. Does that sound like your issue? If so, it's in progress right now. The goal behavior would be to rejoin the live stream at the moment closest to when you stopped playback.

Reloading the same stream again does seem viable as a workaround for the moment but we agree, it should work out-of-the-box.

**Comment by Tralapo** [comment]
@dmlap You say "The goal behavior would be to rejoin the live stream at the moment closest to when you stopped playback", but I don't think that is logical behavior for a live stream. Rejoining at the moment closest to when you stopped, isn't live. In my opinion, it should just jump to the most recent part: it's live.

**Comment by ogzpub** [comment]
@dmlap That's the issue exactly. Glad to hear it's in the works. 

@Tralapo Reloading the source should solve your issue for now. Something like this. 
```
player.on("pause", function () {
  player.one("play", function () {
    if (player.techName_.toLowerCase() == 'flash') {
      player.load();
      player.play();
    } else if (player.techName_.toLowerCase() == 'html5') {
      player.src(sources);
      player.play();
    }      
  });
}); 
```
I don't know if there's a better way to determine the current tech type I just did this real quick since it will go away soon anyway once a true fix is available.

**Comment by Tralapo** [comment]
@ogzpub Does not work

**Comment by ogzpub** [comment]
@Tralapo That's odd, it does work for my particular setup. What happens when it tries to resume? I should have clarified that you would need to replace "sources" in the player.src(sources) line with your source data. In my case sources contains an array such as the one below.
```
var sources = [
  { type: "video/mp4", src: "http://www.example.com/path/to/video.mp4" },
  { type: "video/webm", src: "http://www.example.com/path/to/video.webm" },
  { type: "video/ogg", src: "http://www.example.com/path/to/video.ogv" }
];
```

**Comment by dmlap** [comment]
Resuming live streams should be fixed in [1.2.2](https://github.com/videojs/videojs-contrib-hls/releases/tag/v1.2.2). Please give it a shot and let us know you're still seeing issues.
